### DISCUSSION TITLE
Rollback expected when MigrationUp throw an Exception

### DISCUSSION THREAD

**Discussion by jeremy-c87** [fluentmigrator/fluentmigrator#1954] [Q&A]
Hello,

I am playing a bit with FluentMigrator. I use runner.MigrateUp() to apply modification in my db as explained in the documentation. When the migration code is correct everything works as expected but then I put an error in my SQL, for exemple a wrong column name in a ForeignKey constraint, The Exception is thrown and the migration is simply stopped. I expected a rollback of the current migration (class) at least but apparently not.
Is that expected? How should I write the migration so all the change MigrationUp has down is rolled back like in a transaction?

Thanks.
Best regards.

EDIT 1 :
I think I have undestood a part of the problem. In MariaDB (I use MySql5 as it is a fork), DDL statements create new transaction every statement so this cannot work.
But the question remain. How should I handle the case properly?

**Comment by jeremy-c87** [comment]
Is the project dead? Or the dev(s) are very busy?
I would be glad for any solution for my problem.

Best regards

**Comment by PhenX** [comment]
@jeremy-c87 no the project is not dead, sorry.
To answer you question : Fluent Migrator can only rollback to previous state when the database can handle many DDL statements inside the same transaction, so that excludes MySQL, MariaDB and Oracle (see https://www.datasunrise.com/professional-info/ddl-in-transactions/)

I suggest you make a migration class per DDL statement

**Comment by jeremy-c87** [comment]
Ok, I understand.
Maybe an information regarding this could be visible in the documentation.
Also, an information regarding this behaviour in the exception could be nice.

Best regards
